### ISSUE TITLE
No module named 'transformers' on initial Arm MacOS setup

### ISSUE THREAD

**Issue by highfiiv** [invoke-ai/InvokeAI#650]
**Describe your environment**
MacOS MBP 2021
ZSH
Anaconda
Python (latest version)

Just ran through the Mac Arm installation instructions and cannot get past this error. 

```
(ldm) ➜  stable-diffusion git:(main) python scripts/preload_models.py
Traceback (most recent call last):
  File "/Users/comp/dev/stable-diffusion/scripts/preload_models.py", line 6, in <module>
    from transformers import CLIPTokenizer, CLIPTextModel
ModuleNotFoundError: No module named 'transformers'
```
Has anyone experienced this or know what went wrong?

**Comment by netsvetaev** [comment]
As I remember, I just install them from conda.

**Comment by moelfus** [comment]
@netsvetaev  thanks that helped. I've had the same issue: 
```
RuntimeError: Failed to import transformers.models.clip.modeling_clip because of the following error (look up to see its traceback):
dlopen(/Users/$USER/miniconda3/envs/ldm/lib/python3.10/site-packages/google/protobuf/pyext/_message.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace (__ZN6google8protobuf15FieldDescriptor12TypeOnceInitEPKS1_)
```
solved it by installing transformers manually: 
```
conda install transformers
```
However, I switched to pytorch nightly before, don't know if that makes a difference: 
```
conda install pytorch torchvision torchaudio -c pytorch-nightly
```

**Comment by wukerplank** [comment]
Hm, I checked out the `develop` branch recreated the conda env, but I still get the `No module named 'transformers'` error.

**Comment by mauwii** [comment]
That's strange, because about 3-4 hours ago I completley removed my conda env and the anaconda virtual environment from pyenv and reinstalled it from the ground, had no problems 😮

Did you uninstall protobuf via `brew uninstall protobuf`?

**Comment by wukerplank** [comment]
I just retried it and made sure the brew protobuf is also uninstalled. Still get the same error. I'm an absolute beginner working with these tools, so I suck at debugging this. Sorry to not be of more help!

**Comment by mauwii** [comment]
Have you tried to clean the Repository

```bash
git clean -f
conda clean \
  --yes \
  --all
```

and to reset the conda base env

```bash
conda update \
  --force-reinstall \
  -y \
  -n base \
  -c defaults conda
```

**Comment by wukerplank** [comment]
I ran those commands, but they didn't help.

**Comment by mauwii** [comment]
is rust installed via brew?

can you do a `pip list | grep "protobuf"` and a `conda list -n ldm | grep "transformers"`

my results:

```bash
pip list | grep "protobuf"
protobuf                                  3.20.1

conda list -n ldm | grep "transformers"
taming-transformers       0.0.1                     dev_0    <develop>
transformers              4.21.2                   pypi_0    pypi
```

**Comment by wukerplank** [comment]
Yes, I have Rust via brew. `pip list | grep "protobuf"` seems to be much higher than your version, not sure if that is a problem:

```
protobuf                      4.21.6
```

`conda list -n ldm | grep "transformers"` gets me this:

```
CondaError: Expected exactly one `egg-info` directory in '/Users/wukerplank/Desktop/stable-diffusion', via egg-link 'lib/python3.10/site-packages/stable-diffusion.egg-link'. Instead found: ('stable_diffusion.egg-info', 'latent_diffusion.egg-info').  These are often left over from legacy operations that did not clean up correctly.  Please remove all but one of these.
```

I guess at this point I've unsuccessfully followed so many different guides that my environment is totally messed up 😭

**Comment by mauwii** [comment]
remove latent_diffusion.egg-info (I also only have stable_diffusion.egg-info)

**Comment by wukerplank** [comment]
Now, with `conda list -n ldm | grep "transformers"` I get:

```
taming-transformers       0.0.1                     dev_0    <develop>
transformers              4.21.2                   pypi_0    pypi
```

**Comment by wukerplank** [comment]
Alright, thank you! I'll take another stab at following the guide. 🤞

**Comment by moelfus** [comment]
I also installed stable diffusion yesterday from scratch following the guide. After installing you will have to close and reopen the shell, otherwise conda will fail. The conda installation mentions that, but the guide not ... and first time I overlooked the message. I also switched to pytorch nightly, before figuring out that `transformers` was missing. Afterwards I got a protobuf error. And overcame it by `conda update protobuf`. I don't know if that causes any side effects, but I could successfully use SD

**Comment by burakkurkcu** [comment]
> can you do a `pip list | grep "protobuf"` and a `conda list -n ldm | grep "transformers"`
> 
> my results:
> 
> ```shell
> pip list | grep "protobuf"
> protobuf                                  3.20.1

as you can see protobuf now installs 3.20+ version and there is related github issue on protobuf github regarding that version: https://github.com/protocolbuffers/protobuf/issues/10571 

solution seems as simple as downgrading to some other version, and for my case I just downgraded protobuf to 3.19.4 version by running this command:

`pip install protobuf==3.19.4`

and after that completes, I continued by running this command again:

`python scripts/preload_models.py`

and it completed successfully and the issue seems solved for me.

**Comment by mauwii** [comment]
is protobuf installed via brew as well? @burakkurkcu 

```bash
conda list protobuf -n ldm
# packages in environment at /Users/mauwii/.pyenv/versions/anaconda3-2022.05/envs/ldm:
#
# Name                    Version                   Build  Channel
libprotobuf               3.20.1               hb5ab8b9_4    conda-forge
protobuf                  3.20.1          py310hb07a4bc_0    conda-forge

pip list | grep "protobuf"
protobuf                                  3.20.1
```

I don't have protobuf installed via brew (`brew uninstall protobuf`) and it is working pretty fast with version 3.20.1 👨🏻‍💻

**Comment by asimilon** [comment]
I ran into this same issue; `conda install transformers` didn't solve the issue.  I did not try uninstalling the brew version (since the install instructions for macOS install protobuf via brew), but downgrading protobuf as suggested by [burakkurkcu](https://github.com/burakkurkcu) worked.

**Comment by arice** [comment]
I just ran into this but realized I had rebooted and had not activated my conda environment with `conda activate ldm`

**Comment by 19wolf** [comment]
Not sure if this helps anyone, but in the instructions it says run `python ./scripts/preload_models.py` which didn't work for me, but running `./scripts/preload_models.py` worked fine

**Comment by Christopher-Hayes** [comment]
Having this issue on Ubuntu 22.04 as well.

**Comment by tkolarik** [comment]
for me what fixed the issue was `pip install protobuf==3.19.4` to downgrade

**Comment by psychedelicious** [comment]
This is a bit jumbled for an issue and it looks like most issues were resolved on macOS. Closing; if you still have a problem like this, please create a new issue for focused troubleshooting.
